services/src/main/java/org/keycloak/authentication/authenticators/browser/WebAuthnAuthenticator.java

```diff
@@ -265,7 +265,7 @@ private void setErrorResponse(AuthenticationFlowContext context, final String er
             context.getEvent()
                 .detail(ERR_LABEL, errorCase)
                 .error(Errors.INVALID_USER_CREDENTIALS);
-            errorResponse = createErrorResponse(context, errorCase);
+            errorResponse = createErrorResponse(context, errorCase, errorMessage);
```

Comment:
Question: It seems the intention of this is to show some additional error detail in the UI to the user, is it correct?

I see that it can be useful, on the other hand this can theoretically open potential for XSS attacks. For example if previously there was error displayed in the UI like `Invalid authenticator` and now there is error like `Invalid authenticator: my-webauthn-label` it means the user label is now displayed in the UI error message. In case user label (which for example was filled by "malicious user administrator" in the admin console) is something like `my-label</span><script>alert('some-javascript')</script>` , it can cause that javascript is triggered when some error message is shown.

I am not sure if this can be real attack, maybe error messages are somehow escaped. Just asking if you considered/tested this.

Hint: For the case when you added error messages to the UI for testing purposes, you can instead use `AssertEvents` stuff in the tests. It seems that `errorMessage` is always added as a detail to the error event, so I believe it should be possible to check if error detail has requested value when you're testing error scenarios.

Reply:
@mposolda You're right! It'd be better to remove it for now. And then in some follow-up task to escape the error message and provide it to the user.

Reply:
Ok, Thanks